server/src/main/java/org/opensearch/index/shard/StoreRecovery.java

```diff
@@ -508,6 +516,19 @@ private void recoverFromRemoteStore(IndexShard indexShard) throws IndexShardReco
         }
     }
 
+    private void syncTranslogFilesFromRemoteTranslog(IndexShard indexShard, Repository repository) throws IOException {
+        assert repository instanceof BlobStoreRepository : "repository should be instance of BlobStoreRepository";
+        BlobStoreRepository blobStoreRepository = (BlobStoreRepository) repository;
+        FileTransferTracker fileTransferTracker = new FileTransferTracker(shardId);
+        TranslogTransferManager translogTransferManager = RemoteFsTranslog.buildTranslogTransferManager(
+            blobStoreRepository,
+            indexShard.getThreadPool().executor(ThreadPool.Names.TRANSLOG_TRANSFER),
+            shardId,
+            fileTransferTracker
+        );
+        RemoteFsTranslog.download(translogTransferManager, indexShard.shardPath().resolveTranslog());
+    }
+
```

Comment:
There seems to be a problem here
1. There will be two different instances of `FileTransferTracker` one created here and one created by `RemoteFsTranslog ` which will cause inconsistency in file tracking.
2. The approach looks like a quick fix to workaround the dependency management and instance creation.

Reply:
> The approach looks like a quick fix to workaround the dependency management and instance creation.

Yes, agree. This was required as `RemoteFsTranslog.download` is a static method. It needs to be static as, at some places in the OpenSearch code, we refer to translog files without using Translog class. In the long term, we need to change this behavior and make sure translog files are always accessed via Translog object.

> There will be two different instances of FileTransferTracker one created here and one created by RemoteFsTranslog which will cause inconsistency in file tracking.

The `FileTransferTracker` instance we create here will not be used to track the upload of files, the one created in RemoteFsTranslog will be used. This is not ideal way to handle it but will be fixed with the solution proposed in the first reply above.

Reply:
Created a tracking issue: https://github.com/opensearch-project/OpenSearch/issues/5679